src/api/dependency-discovery/src/dependency-list.js

```diff
@@ -1,14 +1,52 @@
 const { readFile } = require('fs/promises');
 const { join } = require('path');
 const { cwd } = require('process');
+const { getPackument } = require('query-registry');
+
+const dependenciesGlobal = {};
 
 async function getDependencies() {
-  const dependencies = await readFile(join(cwd(), 'deps.txt'), 'utf8');
+  if (Object.keys(dependenciesGlobal).length === 0) {
+    const dependencyList = await readFile(join(cwd(), 'deps.txt'), 'utf8');
+
+    // The order of the alternatives is important!
+    // The regex engine will favor the first pattern on an alternation
+    // even if the other alternatives are subpatterns
+    for (const dependencyName of dependencyList.split(/\r\n|\n|\r/).filter((line) => line !== '')) {
+      dependenciesGlobal[dependencyName] = null;
+    }
+  }
+
+  return dependenciesGlobal;
+}
+
+async function getDependencyList() {
+  const dependencies = await getDependencies();
+  return Object.keys(dependencies);
+}
+
+async function getNpmPackageInfo(packageName) {
+  if (!(await isPackageDependency(packageName))) {
+    return null;
+  }
+
+  const dependencies = await getDependencies();
+
+  if (dependencies[packageName] === null) {
+    const { id, license, gitRepository } = await getPackument({ name: packageName });
+    dependencies[packageName] = { id, license, gitRepository };
+  }
+
+  return dependencies[packageName];
+}
 
-  // The order of the alternatives is important!
-  // The regex engine will favor the first pattern on an alternation
-  // even if the other alternatives are subpatterns
-  return dependencies.split(/\r\n|\n|\r/).filter((line) => line !== '');
+async function isPackageDependency(packageName) {
+  const dependencies = await getDependencies();
+  return dependencies.hasOwnProperty(packageName);
```

Comment:
Why not
```suggestion
  return !!dependencies[packageName]
```

Reply:
There is a slight difference between `!!dependencies[packageName]` and `hasOwnProperty()`. `hasOwnProperty()` returns true if the string passed is a key of the object, while `!!dependencies[packageName]` returns a truth if `dependencies[packageName]` is truthy. If `dependencies[packageName] === null`, then `!!dependencies[packageName]` would be false, and `hasOwnProperty()` is true.

Reply:
I see, you are caching the package info. Still, eslint will complain about this, eslint is not working at the moment.

Reply:
I checked the [available set of lint rules](https://eslint.org/docs/rules/), and none of them really disallow the use of this function. There is only one rule, but it only applies if we use ES2022, which we don't, I believe.

Reply:
It is an interesting rule, for example, if we have a dependency called "hasOwnProperty", when you call `dependencies.hasOwnProperty`, our server will crash because `hasOwnProperty` function is shadowed by the hasOwnProperty key, and it is not a function.

It's this rule https://eslint.org/docs/rules/no-prototype-builtins, valid from ECMAScript 5.1.